dspot/src/main/java/eu/stamp_project/utils/TypeUtils.java

```diff
@@ -93,4 +97,25 @@ public static boolean isString(CtTypeReference type) {
 		}
 		return false;
 	}
+
+	/**
+	 * Returns the full Java qualified name of a method
+	 * @param method
+	 * @return qualified name
+	 */
+	public static String getQualifiedName (CtMethod method) {
```

Comment:
You can write something like: 

```java
StringBuilder qualifiedName = new StringBuilder();
		if (method != null) {
		    if (method.getParent(CtClass.class) != null) {
		        qualifiedName.append(method.getParent(CtClass.class).getQualifiedName())
                        .append(".");
            }
            qualifiedName.append(method.getSimpleName());
        }
        return qualifiedName.toString();
```
`method.getParent(CtClass.class)` will go throught the AST using the `getParent()` accessor until it found a CtClass.
`getQualifiedName()` returns the full qualified name of a class, _e.g._ `my.package.MyClass`.

Reply:
Thanks for the advice. I adopted the getParent (class) signature, but the getQualifiedName() applied for CtClass does not give the qualified name as you affirmed.

Reply:
That's quite strange... I'm gonna have look at this.